src/vs/workbench/services/themes/electron-browser/themeCompatibility.ts

```diff
@@ -26,12 +26,18 @@ function addSettingMapping(settingId: string, colorId: string) {
 
 export function convertSettings(oldSettings: ITokenColorizationRule[], resultRules: ITokenColorizationRule[], resultColors: IColorMap): void {
 	for (let rule of oldSettings) {
-		resultRules.push(rule);
-		if (!rule.scope) {
+		resultRules.push(rule); // push the rule
+		if (!rule.scope) { // if it doesn't have a scope...?? why?
 			let settings = rule.settings;
 			if (!settings) {
-				rule.settings = {};
+				rule.settings = {}; // make settings an empty object if it doesn't exist
 			} else {
+				// * For each key in settings (foreground, background, fontstyle)
+				// * If the key is in a settingToColorIdMapping, get the color from the setting
+				//   then find all the colorIds mapped to that setting, and set those to the color
+				//   It is not yet clear to me what this looks like in practice
+				// * Remove all keys that aren't foreground or background
+				//   Why remove fontstyle?
```

Comment:
When I go back in to remove my "question" comments, should I then also change `if (key !== 'foreground' && key !== 'background')` to `if (key !== 'foreground' && key !== 'background' && key !== 'fontStyle')`?